plugins/node/opentelemetry-instrumentation-aws-sdk/src/aws-sdk.ts

```diff
@@ -400,12 +400,16 @@ export class AwsInstrumentation extends InstrumentationBase<AwsSdkInstrumentatio
                     request: normalizedRequest,
                     requestId: requestId,
                   };
-                  self.servicesExtensions.responseHook(
+                  const override = self.servicesExtensions.responseHook(
                     normalizedResponse,
                     span,
                     self.tracer,
                     self.getConfig()
                   );
+                  if (override) {
+                    response.output = override;
+                    normalizedResponse.data = override;
+                  }
                   self._callUserResponseHook(span, normalizedResponse);
```

Comment:
I considered whether this should be done for the user hook too but didn't think there's enough use case for it. Currently the change is only internal since AFAIK, users can't define service extensions